c10/cuda/CUDACachingAllocator.h

```diff
@@ -180,13 +182,15 @@ struct TraceEntry {
       size_t size,
       cudaStream_t stream,
       approx_time_t time,
-      std::shared_ptr<GatheredContext> context = nullptr)
+      std::shared_ptr<GatheredContext> context = nullptr,
+      std::string name = "")
```

Comment:
I'd also like to avoid adding new fields to this struct that only are used for custom annotations. Is it possible to put the name into the GatheredContext object, which already holds information about "where" this event occurred For instance, we can add an additional "frame" to the top of the stack trace. That would make these events easier to handle in viz as well, since the data would get threaded into the frames object. Otherwise there will be a ton of places that would have to be modified to get the name field threaded through to the right parts to visualize.

Reply:
This is a great idea! I've updated the PR to save the name and stage (START or END) into a frame in the GatheredContext. This makes sense to me! Thanks!